pkg/config/manager.go

```diff
@@ -0,0 +1,66 @@
+package config
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"io/ioutil"
+	"os"
+
+	"gopkg.in/yaml.v2"
+	"k8s.io/klog/v2"
+)
+
+// Manager controls the configuration file loading, and can assert if it has changed on disk.
+type Manager struct {
+	cfgFilePath string // path to configuration file
+	md5sum      string // md5sum of the initial content
+}
+
+// ConfigHasChanged checks the current configuration contents, comparing with that it has been
+// instantiated with.
+func (m *Manager) ConfigHasChanged() bool {
+	// given the md5sum is not yet set, the configuration payload won't be marked as changed
+	if m.md5sum == "" {
+		return false
+	}
+
+	// reading the configration file payload again and comparing with the md5sum stored, when there
+	// are errors reading the file, it does not mark the configuration as changed
+	payload, err := ioutil.ReadFile(m.cfgFilePath)
+	if err != nil {
+		klog.Errorf("Reading configuration-file '%s': '%#v'", m.cfgFilePath, err)
+		return false
+	}
+	sum := md5.Sum(payload)
+	return m.md5sum != hex.EncodeToString(sum[:])
+}
+
+// LoadConfig read the local configuration file, make sure the current contents are summed, so we can
+// assert if there are changes later on.
+func (m *Manager) LoadConfig() (*Config, error) {
```

Comment:
OK seeing all this, I think this lends itself to my option 1) in the comment I made in main.go @otaviof 

WDYT?